Retry/src/main/java/org/techsalad/retry/service/SaladsService.java

```diff
@@ -0,0 +1,23 @@
+package org.techsalad.retry.service;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.springframework.stereotype.Service;
+
+@Service
+public class SaladsService {
+
+  private int failuresCount = 0;
```

Comment:
as a bonus/optional step, we might want to refactor the sample in a way, that failing service would be part of unit test only.

as this means, we'll have to redeploy to demo again to simulate the failure.

potentially we should just call some external URL in the source code, that would be mocked for the test/demo purposes. Wiremock might be an option: https://wiremock.org/docs/stateful-behaviour/